module/documents/actor/actor.mjs

```diff
@@ -1962,6 +2059,10 @@ export default class Actor5e extends Actor {
       newTokenData._id = t.id;
       newTokenData.actorId = newActor.id;
       newTokenData.actorLink = true;
+
+      const dOriginalActorData = foundry.utils.getProperty(d, "flags.dnd5e.originalActor");
```

Comment:
`originalActor` should just be an `id`, rather than some object of data, no?

```suggestion
      const dOriginalActor = foundry.utils.getProperty(d, "flags.dnd5e.originalActor");
```